### PR TITLE
Add check_connection flag to SourceRequest

### PR DESCRIPTION
**Pull request by ganisimov**
Related to #2359

### CODE REVIEW

File: src/main/java/org/ohdsi/webapi/service/converters/SourceRequestToSourceConverter.java [hunk 1]
```diff
@@ -29,6 +29,9 @@ public Source convert(SourceRequest request) {
         source.setKeyfileName(request.getKeyfileName());
         source.setKrbAdminServer(request.getKrbAdminServer());
         source.setKrbAuthMethod(KerberosAuthMechanism.getByName(request.getKrbAuthMethod()));
+        if (null != request.isCheckConnection()) {
```
Comment (anthonysena) [inline src/main/java/org/ohdsi/webapi/service/converters/SourceRequestToSourceConverter.java:32]: @ganisimov can you change the evaluation such that is reads: request.isCheckConnection() != null? The other way is a bit confusing to me at least.

File: src/main/java/org/ohdsi/webapi/source/SourceController.java [hunk 2]
```diff
@@ -238,6 +238,9 @@ public SourceInfo updateSource(@PathParam("sourceId") Integer sourceId, @FormDat
       }
       setKeyfileData(updated, source, file);
       transformIfRequired(updated);
+      if (null == request.isCheckConnection()) {
```
Comment (anthonysena) [inline src/main/java/org/ohdsi/webapi/source/SourceController.java:241]: Same for this evaluation

**Review by anthonysena — CHANGES_REQUESTED** [review]
Looks good with a few minor comments about the `if` statements

File: src/main/java/org/ohdsi/webapi/service/converters/SourceRequestToSourceConverter.java (hunk 1 above)
Comment (ganisimov) [reply src/main/java/org/ohdsi/webapi/service/converters/SourceRequestToSourceConverter.java]: Agree, this doesn't make any sense for Java. Thanks for looking into this!

**Comment by chrisknoll** [comment]
Looks good to me, thanks for addressing those comments.

**Pull request merged**